### ISSUE TITLE
Nx Micro Frontend installation fails out of the box

### ISSUE THREAD

**Issue by DonTomato** [nrwl/nx#7905]
Tried to setup a Micro Frontend with Angular and Nx according to the [Nx guide](https://nx.dev/l/a/guides/setup-mfe-with-angular). After following all instructions firstly I get the following error when run:
![image](https://user-images.githubusercontent.com/1098084/143581015-817b4cf5-a425-4ae8-bd15-11d61717897e.png)

It seems I can fix it with installing `webpack` packages manually:
```
npm install --save-dev webpack 
npm install --save-dev webpack-dev-server
```
But now after I run the dashboard shell app(`nx run dashboard:serve-mfe`) it started to be reloading constantly with the following error:
![image](https://user-images.githubusercontent.com/1098084/143581361-8abac4be-64cf-4986-95b3-1d88f3b10934.png)
BTW: `remoteEntry.js` is accessible by ` http://localhost:5000/remoteEntry.js` actually.

Repo: [https://github.com/DonTomato/ng-mfe](https://github.com/DonTomato/ng-mfe)

### Environment
```
λ nx report                                                                                                                                                                                           
                                                                                                                                                                                                      
>  NX  Report complete - copy this into the issue template                                                                                                                                            
                                                                                                                                                                                                      
  Node : 16.13.0                                                                                                                                                                                      
  OS   : win32 x64                                                                                                                                                                                    
  npm  : 8.1.0                                                                                                                                                                                        
                                                                                                                                                                                                      
  nx : 13.2.2                                                                                                                                                                                         
  @nrwl/angular : 13.2.2                                                                                                                                                                              
  @nrwl/cli : 13.2.2                                                                                                                                                                                  
  @nrwl/cypress : 13.2.2                                                                                                                                                                              
  @nrwl/devkit : 13.2.2                                                                                                                                                                               
  @nrwl/eslint-plugin-nx : 13.2.2                                                                                                                                                                     
  @nrwl/express : undefined                                                                                                                                                                           
  @nrwl/jest : 13.2.2                                                                                                                                                                                 
  @nrwl/linter : 13.2.2                                                                                                                                                                               
  @nrwl/nest : undefined                                                                                                                                                                              
  @nrwl/next : undefined                                                                                                                                                                              
  @nrwl/node : undefined                                                                                                                                                                              
  @nrwl/nx-cloud : undefined                                                                                                                                                                          
  @nrwl/react : undefined                                                                                                                                                                             
  @nrwl/react-native : undefined                                                                                                                                                                      
  @nrwl/schematics : undefined                                                                                                                                                                        
  @nrwl/tao : 13.2.2                                                                                                                                                                                  
  @nrwl/web : undefined                                                                                                                                                                               
  @nrwl/workspace : 13.2.2                                                                                                                                                                            
  @nrwl/storybook : 13.2.2                                                                                                                                                                            
  @nrwl/gatsby : undefined                                                                                                                                                                            
  typescript : 4.4.4                                                                                                                                                                                  
  rxjs : 7.4.0                                                                                                                                                                                        
  ---------------------------------------                                                                                                                                                             
  Community plugins:                                                                                                                                                                                  
         @angular-architects/module-federation: 12.5.3                                                                                                                                                
         @angular/animations: 13.0.2                                                                                                                                                                  
         @angular/common: 13.0.2                                                                                                                                                                      
         @angular/compiler: 13.0.2                                                                                                                                                                    
         @angular/core: 13.0.2                                                                                                                                                                        
         @angular/forms: 13.0.2                                                                                                                                                                       
         @angular/platform-browser: 13.0.2                                                                                                                                                            
         @angular/platform-browser-dynamic: 13.0.2                                                                                                                                                    
         @angular/router: 13.0.2                                                                                                                                                                      
         @angular-devkit/build-angular: 13.0.3                                                                                                                                                        
         @angular/compiler-cli: 13.0.2                                                                                                                                                                
         @angular/language-service: 13.0.2                                                                                                                                                            
                                                                                                                                                                                                      
(node:11872) [DEP0148] DeprecationWarning: Use of deprecated folder mapping "./" in the "exports" field module resolution of the package at C:\Projects\Waste\ng-mfe\node_modules\tslib\package.json. 
Update this package.json to use a subpath pattern like "./*".                                                                                                                                         
(Use `node --trace-deprecation ...` to show where the warning was created)                                                                                                                            
```

**Comment by DonTomato** [comment]
It seems the problem with missing `webpack`, appeared with `13.0.0` release.
But problem with reloading occurs with some latest release, after `13.2.x`

**Comment by avington** [comment]
I am seeing the same reloading issue after running nx migrate from "@nrwl/workspace": "13.1.4" to "@nrwl/workspace": "13.2.3"

**Comment by namdien177** [comment]
Facing this issue right now. I can fix the NPM install error by `npm i --legacy-peer-deps` but the  `serve-mfe` command makes the browser constantly reload.

**Comment by schackito** [comment]
looks like it's a general problem:
https://github.com/nrwl/nx/pull/7940

**Comment by Coly010** [comment]
It's related to Angular 13 switching to ESM.

Fix is being worked on: https://github.com/nrwl/nx/pull/8020

**Comment by Coly010** [comment]
#8080 has been merged. It should be released with Nx 13.3. Once it does, try it out and let me know how it goes.

It's recommended to use Yarn when using MFEs with Angular due to Webpack version hoisting

**Comment by DonTomato** [comment]
@Coly010 It still doesn't work with different error this time: `Uncaught TypeError: Failed to resolve module specifier "login@http://localhost:5001/loginEntry.js". Relative references must start with either "/", "./", or "../"`
